crates/provider/src/ext/erc4337.rs

```diff
@@ -0,0 +1,217 @@
+pub trait Erc4337Api<N, T>: Send + Sync {
+    /// Sends a user operation to the bundler, as defined in ERC-4337.
+    ///
+    /// Entry point changes based on the user operation type.
+    async fn send_user_operation(
+        &self,
+        user_op: SendUserOperation,
+        entry_point: Address,
+    ) -> TransportResult<SendUserOperationResponse>;
+
+    /// Returns the list of supported entry points.
+    async fn supported_entry_points(&self) -> TransportResult<Vec<Address>>;
+
+    /// Returns the receipt for any user operation.
+    ///
+    /// Hash is the same returned by any user operation.
+    async fn get_user_operation_receipt(
+        &self,
+        user_op_hash: Bytes,
+    ) -> TransportResult<UserOperationReceipt>;
+
+    /// Estimates the gas for a user operation.
+    ///
+    /// Entry point changes based on the user operation type.
+    async fn estimate_user_operation_gas(
+        &self,
+        user_op: SendUserOperation,
+        entry_point: Address,
+    ) -> TransportResult<UserOperationGasEstimation>;
+}
+
+#[cfg_attr(target_arch = "wasm32", async_trait::async_trait(?Send))]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait::async_trait)]
+impl<N, T, P> Erc4337Api<N, T> for P
+where
+    N: Network,
+    T: Transport + Clone,
+    P: Provider<T, N>,
+{
+    async fn send_user_operation(
+        &self,
+        user_op: SendUserOperation,
+        entry_point: Address,
+    ) -> TransportResult<SendUserOperationResponse> {
+        match user_op {
+            SendUserOperation::EntryPointV06(user_op) => {
+                self.client().request("eth_sendUserOperation", (user_op, entry_point)).await
+            }
+            SendUserOperation::EntryPointV07(packed_user_op) => {
+                self.client().request("eth_sendUserOperation", (packed_user_op, entry_point)).await
+            }
+        }
+    }
+
+    async fn supported_entry_points(&self) -> TransportResult<Vec<Address>> {
+        self.client().request("eth_supportedEntryPoints", ()).await
+    }
+
+    async fn get_user_operation_receipt(
+        &self,
+        user_op_hash: Bytes,
+    ) -> TransportResult<UserOperationReceipt> {
+        self.client().request("eth_getUserOperationReceipt", (user_op_hash,)).await
+    }
+
+    async fn estimate_user_operation_gas(
+        &self,
+        user_op: SendUserOperation,
+        entry_point: Address,
+    ) -> TransportResult<UserOperationGasEstimation> {
+        match user_op {
+            SendUserOperation::EntryPointV06(user_op) => {
+                self.client().request("eth_estimateUserOperationGas", (user_op, entry_point)).await
+            }
+            SendUserOperation::EntryPointV07(packed_user_op) => {
+                self.client()
+                    .request("eth_estimateUserOperationGas", (packed_user_op, entry_point))
+                    .await
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
these tests don't do anything, please remove them 
don't even think this is supported on vanilla geth so we can't really test this using geth

Reply:
I agree! Removed.